addon/-private/modifier.ts

```diff
@@ -541,11 +543,19 @@ export default class ElementQueryModifier extends Modifier<ModifierArgs> {
       : sizeObjectsSortedAsc[sizeObjectsSortedAsc.length - 1];
   }
 
+  _maybeRunPromiseResolveHasBeenInstalled(): void {
+    if (this._promiseResolveHasBeenInstalled) {
+      this._promiseResolveHasBeenInstalled();
+      this._promiseResolveHasBeenInstalled = undefined;
+    }
+  }
+
   _didResizeHandler(): void {
     window.requestAnimationFrame(() => {
       if (!this.args.named.isDisabled && !this.isDestroying && !this.isDestroyed) {
         this.applyAttributesToElement();
         this.callOnResize();
+        this._maybeRunPromiseResolveHasBeenInstalled();
```

Comment:
What if this is called after the modifier is already destroyed, wouldn't this never resolve the promise, i.e. make tests hang? Can we call `this._maybeRunPromiseResolveHasBeenInstalled?.();` (note the optional chaining) outside of the if-block?

Reply:
Done.